src/graph/graph.c

```diff
@@ -370,21 +289,26 @@ void Graph_DeleteNodes(Graph *g, NodeID *IDs, size_t IDCount) {
     for(int j = 0, i = 0; i < IDCount; i++) {
         int id = IDs[i];
         // Require a replacement?
-        if(id < (g->node_count - IDCount)) {
+        if(id < (Graph_NodeCount(g) - IDCount)) {
             // Locate a valid replacement.
             while(replacements[j].delete) j++;
             replacements[j++].nodeID = id;
         }
     }
 
     /* Replace removed nodes within node blocks. */
+    size_t deletedNodeCount = 0;
     for(int j = 0, i = 0; i < IDCount; i++) {
         Replacement r = replacements[i];
-        // No need to perform replacement.
-        if(r.delete) continue;
-        // Override nodeID with replacementID.
-        _Graph_NodeBlockMigrateNode(g, r.replacementID, r.nodeID);
+        if(r.delete) {
+            // No need to perform replacement.
+            deletedNodeCount++;
+        } else {
+            // Override nodeID with replacementID.
+            DataBlock_MigrateItem(g->nodes, r.replacementID, r.nodeID);
```

Comment:
It's probably not worth digging too deep on delete ops right now, but some of the delete issues stem from this migrate loop occurring before `_Graph_MigrateRowCol` (317). This call will reduce `g->nodes->itemCount`, which in turn causes matrices to get resized before the to-be-swapped column is put into the to-be-deleted position.